KenticoCloud.Delivery/DeliveryClient.cs

```diff
@@ -521,5 +525,24 @@ public DeliveryClient(string projectId, string previewApiKey) : this(new Deliver
 
             throw new DeliveryException(response.StatusCode, await response.Content.ReadAsStringAsync());
         }
+
+        private bool TryGetContentTypeCodename(Type contentType, out string codename)
```

Comment:
I'd definitely put the whole thing in a separate class and unit-test it.
Also, consider making it an extension method. `parameters.ApplyTypeParameters<T>()`. But that's just a matter of taste.

Reply:
Extracted into separate class which was unit tested. We can do it as extension method later.